torchrl/data/rlhf/dataset.py

```diff
@@ -0,0 +1,467 @@
+            based on a particular condition. Tokenization and other
+            "elementwise" operations on the data are performed by the process
+            function which is mapped over the dataset.
+        root_dir (path, optional): the path where the datasets are stored.
+            Defaults to ``"$HOME/.cache/torchrl/data"``
+        from_disk (bool, optional): if ``True``, :func:`datasets.load_from_disk`
+            will be used. Otherwise, :func:`datasets.load_dataset` will be used.
+            Defaults to ``False``.
+
+    The dataset will be stored in ``root/<split>/<max_length>/``
+    Examples:
+        >>> from torchrl.data.rlhf import TensorDictTokenizer        >>> from torchrl.data.rlhf.reward import  pre_tokenization_hook
+        >>> split = "train"
+        >>> max_length = 550
+        >>> dataset_name = "CarperAI/openai_summarize_comparisons"
+        >>> dataset = create_or_load_dataset(
+        ...     split,
+        ...     max_length,
+        ...     dataset_name,
+        ...     TensorDictTokenizer,
+        ...     pre_tokenization_hook=pre_tokenization_hook,
+        ... )
+        >>> print(dataset)
+        TensorDict(
+            fields={
+                train: TensorDict(
+                    fields={
+                        550: TensorDict(
+                            fields={
+                                attention_mask: MemmapTensor(shape=torch.Size([185068, 550]), device=cpu, dtype=torch.int64, is_shared=False),
+                                input_ids: MemmapTensor(shape=torch.Size([185068, 550]), device=cpu, dtype=torch.int64, is_shared=False)},
+                            batch_size=torch.Size([185068, 550]),
+                            device=None,
+                            is_shared=False)},
+                    batch_size=torch.Size([185068, 550]),
+                    device=None,
+                    is_shared=False)},
+            batch_size=torch.Size([185068, 550]),
+            device=None,
+            is_shared=False)
+
+    """
+    if root_dir is None:
+        root_dir = Path(os.environ.get("HOME")) / ".cache/torchrl/data/"
+        os.makedirs(root_dir, exist_ok=True)
+    root_dir = Path(root_dir)
+    data_dir = root_dir / str(Path(dataset_name).name).split("-")[0]
+    data_dir_total = data_dir / split / str(max_length)
+    # search for data
+    if os.path.exists(data_dir_total):
+        dataset = TensorDict.load_memmap(data_dir_total)
+        return dataset
+    dataset = load_dataset(
+        split=split,
+        dataset_name=dataset_name,
+        pre_tokenization_hook=pre_tokenization_hook,
+        from_disk=from_disk,
+    )
+    dataset = tokenize(
+        dataset,
+        max_length=max_length,
+        tokenizer_fn=tokenizer_fn,
+    )
+    prefix = (split, str(max_length))
+    return dataset_to_tensordict(
+        dataset, data_dir=data_dir, prefix=prefix, valid_mask_key="valid_sample"
+    )[prefix]
+
+
+def load_dataset(
```

Comment:
this is only used internally? should we put a "_" in front?
```suggestion
def _load_dataset(
```

Reply:
Is it? IDK if we want to expose it or not. To me it's a well self-contained function properly documented that could be used somewhere else, but I'm happy to make it private if there's no use.

Reply:
I think in the common case user will want either to load or create. Hence my proposal to keep this internal and expose only the maybe-renamed other function